### PR TITLE
New API for volume/snapshot capacity usage.

### PR DESCRIPTION
**Pull request by talakad**
Returns exclusive and shared capacity in use by a volume/snapshot.

Signed-off-by: veda <veda@portworx.com>

**What this PR does / why we need it**:
Where possible this would show the actual storage in use by volume/snapshot.
**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: api/api.proto [hunk 1]
```diff
@@ -460,6 +460,19 @@ message Stats {
   uint64 interval_ms = 10;
 }
 
+message CapacityUsageInfo {
+  // ExclusiveBytes used by this snap
+  uint64 exclusive_bytes = 1;
+  // SharedBytes used by this snap with other snaps and/or parent
+  uint64 shared_bytes = 2;
+  // TotalByes used by this volume
```
Comment (kschoche) [inline api/api.proto]: typo TotalByes

File: api/api.proto [hunk 2]
```diff
@@ -460,6 +460,19 @@ message Stats {
   uint64 interval_ms = 10;
 }
 
+message CapacityUsageInfo {
+  // ExclusiveBytes used by this snap
+  uint64 exclusive_bytes = 1;
```
Comment (lpabon) [inline api/api.proto]: Use int64's

File: api/api.proto [hunk 3]
```diff
@@ -460,6 +460,19 @@ message Stats {
   uint64 interval_ms = 10;
 }
 
+message CapacityUsageInfo {
+  // ExclusiveBytes used by this snap
```
Comment (lpabon) [inline api/api.proto]: Provide enough information on each documentation to explain what it is used for and how to interpret it for those who do not know much about storage.

File: api/api.proto [hunk 4]
```diff
@@ -460,6 +460,19 @@ message Stats {
...
+  uint64 exclusive_bytes = 1;
+  // SharedBytes used by this snap with other snaps and/or parent
+  uint64 shared_bytes = 2;
+  // TotalBytes used by this volume
+  uint64 total_bytes = 3;
+  // ErrorString indicates any error encountered for excl/shared
+  // fields
+  string error_details = 4;
```
Comment (lpabon) [inline api/api.proto]: This is confusing because errors should be returned in the RPC call instead.

File: api/api.proto [hunk 5]
```diff
@@ -460,6 +460,19 @@ message Stats {
...
+  // IsSnap is set true for shapshots
+  bool is_snap = 5;
```
Comment (lpabon) [inline api/api.proto]: What is the purpose of this value? What would be the use of it?

File: api/api.proto [hunk 6]
```diff
@@ -460,6 +460,19 @@ message Stats {
   uint64 interval_ms = 10;
 }
 
+message CapacityUsageInfo {
```
Comment (lpabon) [inline api/api.proto]: Add a small comment to the message itself

File: api/server/sdk/volume_ops.go [hunk 7]
```diff
@@ -352,6 +352,50 @@ func (s *VolumeServer) Stats(
...
+			err.Error())
+	}
+
+	resp := &api.SdkVolumeCapacityUsageResponse{}
+	resp.CapacityUsageInfo = &api.CapacityUsageInfo{}
+	if dResp != nil {
+		resp.CapacityUsageInfo.ExclusiveBytes = dResp.CapacityUsageInfo.ExclusiveBytes
+		resp.CapacityUsageInfo.SharedBytes = dResp.CapacityUsageInfo.SharedBytes
+		resp.CapacityUsageInfo.TotalBytes = dResp.CapacityUsageInfo.TotalBytes
+		if dResp.ErrorDetails != "" {
```
Comment (lpabon) [inline api/server/sdk/volume_ops.go]: Instead of comparing strings, let's make this an `error` val.

Comment (talakad) [reply api/server/sdk/volume_ops.go]: done.

**Pull request merged**